frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -349,6 +351,36 @@ impl<T: Config> SignedSubmissions<T> {
 	}
 }
 
+/// Type that can be used to calculate the deposit base for signed submissions.
+///
+/// The deposit base is calculated as a geometric progression based on the number of signed
+/// submissions in the queue. The size of the queue represents the progression term.
+pub struct GeometricDepositBase<T> {
+	_marker: PhantomData<T>,
+}
+
+impl<T: Config> SignedDepositBase<BalanceOf<T>> for GeometricDepositBase<T> {
+	/// Increase factor of the geometric progression, as a percentage.
+	///
+	/// The nth progression term will be `IncreaseFactor`% larger than the nth-1 term.
+	type IncreaseFactor = T::SignedDepositBaseIncreaseFactor;
+
+	/// Increase the base deposit as a geometric progression based on the number of signed
+	/// submissions.
+	///
+	/// The nth term is obtained by calculating `base * (1 + increase_factor)^nth`. Example: factor
+	/// 5, with initial deposit of 1000 and 10% of increase factor is 1000 * (1 + 1.1)^5.
```

Comment:
Is the calculation formula correct or may be there is a typo?
```suggestion
	/// 5, with initial deposit of 1000 and 10% of increase factor is 1000 * (1 + 0.1)^5.
```